sycl/test/basic_tests/handler/parallel_for_args.cpp

```diff
@@ -0,0 +1,94 @@
+// This is a basic acceptance test which is intended to check that all valid
+// combinations of arguments to parallel_for are accepted by the implementation
+// Note: cases with 'auto' as an argument type are covered by
+//   handler_generic_lambda_interface.cpp
+//
+// RUN: %clangxx -fsycl -fsyntax-only -ferror-limit=0 -Xclang -verify %s
+// RUN: %if preview-breaking-changes-supported %{%clangxx -fsycl -fsyntax-only -Xclang -verify %s -I %sycl_include -fpreview-breaking-changes%}
+//
+// expected-no-diagnostics
+
+#include <sycl/sycl.hpp>
+
+template <int Dims> struct ConvertibleFromItem {
+  ConvertibleFromItem(sycl::item<Dims>) {}
```

Comment:
Do we need to test it with `item<Dims, false>`?

Reply:
Looks like we may do this as we have check `std::is_convertible_v<item<Dims, false>, LambdaArgType>`

Reply:
There is a `TODO` about adding those checks, but I can add them within this PR right away

Reply:
Note: `item<Dims, false>, kernel_handler` case seems to be broken. I will see if I can quickly fix it, but if not, I would prefer to postpone it to a separate PR, because this one is expected to fix CTS build breakage for some tests

Reply:
Submitted #11679 as a follow-up to add those tests and fix issues with `item<Dims, false>`